### ISSUE TITLE
Simple cov from hash returns an array

### ISSUE THREAD

**Issue by Mziserman** [whitesmith/rubycritic#369]
Hey, 

I think this commit https://github.com/simplecov-ruby/simplecov/commit/9ed35debcd6e5b4a22e99a655c9b40be0d7da142# broke https://github.com/whitesmith/rubycritic/blob/master/lib/rubycritic/analysers/coverage.rb#L44 as it expects a SimpleCov::Result instead of an array.

**Comment by etagwerker** [comment]
@Mziserman Thanks for reporting this! It is a known issue that the most recent version of SimpleCov is not compatible with the integration we have implemented in `rubycritic` -- See: https://github.com/whitesmith/rubycritic/issues/359 

Are you getting an error right now? If that is the case, we might need to add a check to make sure it doesn't break when someone is running `rubycritic` with a more recent version of SimpleCov.

**Comment by Mziserman** [comment]
@etagwerker thanks for answering ! 

I am getting an error : 
```bash
running simple_cov
Traceback (most recent call last):
	14: from /usr/local/bundle/bin/rubycritic:23:in `<main>'
	13: from /usr/local/bundle/bin/rubycritic:23:in `load'
	12: from /usr/local/bundle/gems/rubycritic-4.5.1/bin/rubycritic:10:in `<top (required)>'
	11: from /usr/local/bundle/gems/rubycritic-4.5.1/lib/rubycritic/cli/application.rb:21:in `execute'
	10: from /usr/local/bundle/gems/rubycritic-4.5.1/lib/rubycritic/commands/default.rb:19:in `execute'
	 9: from /usr/local/bundle/gems/rubycritic-4.5.1/lib/rubycritic/commands/default.rb:24:in `critique'
	 8: from /usr/local/bundle/gems/rubycritic-4.5.1/lib/rubycritic/analysers_runner.rb:29:in `run'
	 7: from /usr/local/bundle/gems/rubycritic-4.5.1/lib/rubycritic/analysers_runner.rb:29:in `each'
	 6: from /usr/local/bundle/gems/rubycritic-4.5.1/lib/rubycritic/analysers_runner.rb:32:in `block in run'
	 5: from /usr/local/bundle/gems/rubycritic-4.5.1/lib/rubycritic/analysers/coverage.rb:18:in `run'
	 4: from /usr/local/bundle/gems/rubycritic-4.5.1/lib/rubycritic/core/analysed_modules_collection.rb:32:in `each'
	 3: from /usr/local/bundle/gems/rubycritic-4.5.1/lib/rubycritic/core/analysed_modules_collection.rb:32:in `each'
	 2: from /usr/local/bundle/gems/rubycritic-4.5.1/lib/rubycritic/analysers/coverage.rb:19:in `block in run'
	 1: from /usr/local/bundle/gems/rubycritic-4.5.1/lib/rubycritic/analysers/coverage.rb:32:in `find_coverage_percentage'
/usr/local/bundle/gems/rubycritic-4.5.1/lib/rubycritic/analysers/coverage.rb:44:in `find_source_file': undefined method `source_files' for #<Array:0x0000561aef7a7938> (NoMethodError)
```

I however didn't manage to get the error starting from a blank rails app with rspec

**Comment by valterfernandesdev** [comment]
Same problem here. 
![image](https://user-images.githubusercontent.com/22217090/90557494-f0c12d00-e170-11ea-8579-6350bfa187e1.png)

**Comment by etagwerker** [comment]
This was fixed by #370. Thanks @Mziserman!
